libraries/core/src/main/kotlin/com/bumble/appyx/core/composable/Children.kt

```diff
@@ -140,30 +175,32 @@ class ChildrenTransitionScope<T : Any, S>(
                 }
         }
 
-        val visibleElementsFlow = remember {
+        val screenStateFlow = remember {
             this@ChildrenTransitionScope
                 .navModel
                 .screenState
-                .map { list ->
-                    list
-                        .onScreen
-                        .filter { clazz.isInstance(it.key.navTarget) }
-                }
         }
-        val children by visibleElementsFlow.collectAsState(emptyList())
 
+        val children by screenStateFlow.collectAsState()
 
-        children.forEach { navElement ->
-            key(navElement.key.id) {
-                Child(
-                    navElement,
-                    saveableStateHolder,
-                    transitionParams,
-                    transitionHandler,
-                    block
-                )
+        children
+            .onScreen
+            .filter { clazz.isInstance(it.key.navTarget) }
+            .forEach { navElement ->
+                key(navElement.key.id) {
+                    CompositionLocalProvider(
+                        LocalNodeTargetVisibility provides
```

Comment:
Are there any tests covering this? Should we add some of them? Feels like the `ScreenState` and a few other things could benefit from them? Thoughts?

Reply:
I can add a UI test for that

Reply:
added